yb-voyager/cmd/exportSchema.go

```diff
@@ -195,24 +201,75 @@ func exportSchema() error {
+func runAssessMigrationCmdBeforExportSchemaIfRequired(exportSchemaCmd *cobra.Command) error {
+	if !bool(assessSchemaBeforeExport) {
+		log.Infof("skipping running assess-migration command before export schema as flag --assess-schema-before-export is set as false.")
+		return nil
 	}
 
-	if msr.MigrationAssessmentDone {
+	if source.DBType != POSTGRESQL {
+		log.Infof("skipping running assess-migration command before export schema as source DB type is not PostgreSQL.")
 		return nil
 	}
-	if source.DBType != POSTGRESQL {
-		// At this point, we are only interested in nudging postgres users to run assessment
-		// because it is far more advanced and useful for them.
+
+	if ok, _ := IsMigrationAssessmentDoneDirectly(metaDB); ok {
+		log.Infof("migration assessment is already done, skipping running assess-migration command.")
+		return nil
+	} else if ok, _ := IsMigrationAssessmentDoneViaExportSchema(); ok {
+		log.Infof("migration assessment is already done via export schema, skipping running assess-migration command.")
 		return nil
 	}
-	if !utils.AskPrompt("It is recommended to run assess-migration before exporting schema. https://docs.yugabyte.com/preview/yugabyte-voyager/migrate/assess-migration/ \n" +
-		"Do you want to continue anyway without running assess-migration") {
-		utils.ErrExit("Aborting...")
+
+	var assessFlagsWithValues []string
+	commonFlags := utils.GetCommonFlags(exportSchemaCmd, assessMigrationCmd)
+	for _, flag := range commonFlags {
+		// don't pass start-clean flag to assess-migration command here
+		if flag.Name == "start-clean" || !flag.Changed {
+			continue
+		}
+
+		// bool flags: --flag=value
+		// everything else: --flag value
+		if flag.Value.Type() == "bool" {
+			assessFlagsWithValues = append(assessFlagsWithValues,
+				fmt.Sprintf("--%s=%s", flag.Name, flag.Value.String()),
+			)
+		} else {
+			assessFlagsWithValues = append(assessFlagsWithValues,
+				"--"+flag.Name,
+				flag.Value.String(),
+			)
+		}
 	}
+
+	// Append --yes=true(irrespective) at the end to override any --yes=false if set in export schema cmd
+	assessFlagsWithValues = append(assessFlagsWithValues,
+		"--assessment-invoked-from-export-schema", "true",
+		"--iops-capture-interval", "0", // TODO: any small but significant duration will be better than 0
+		"--target-db-version", ybversion.LatestStable.String(),
+		"--yes",
+	)
+
+	// locate voyager binary
+	voyagerExecutable, err := os.Executable()
+	if err != nil {
+		log.Warnf("cannot locate executable, skipping assessment: %v", err)
+		return nil
+	}
+
+	// Invoke the assess-migration command as a subprocess
+	cmd := exec.Command(voyagerExecutable, append([]string{"assess-migration"}, assessFlagsWithValues...)...)
+	cmd.Stdout = os.Stdout
+
+	var stderrBuf bytes.Buffer
+	cmd.Stderr = &stderrBuf
+
+	// run and ignore exit status
+	if err := cmd.Run(); err != nil {
+		utils.PrintAndLog("failed to assess the schema: %v, continuing with export schema...", err)
```

Comment:
again, here, let's return the error properly. From the perspective of this function, it has encountered an error, and it should return it. The decision to "continue with export schema" is the callers.
- let's warn log the stderr. 
- let's NOT printAndLog in this function (that is better suited to do in the caller function)

Reply:
done, take a look.